### PR TITLE
PLAT-1168: Fix avatar group max display issue

### PR DESCRIPTION
**Pull request by mihaelpalfi**
[JIRA ticket](https://reciprocitylabs.atlassian.net/browse/PLAT-1168)

Fixed the issue. Regarding seperation to another folder was suggested by you guys. :) I think this is not a smart one and i can move it under the components folder, just that we decide.

#### Description
As per JIRA ticket

#### ChangeLOG

- [ ] I have added all notable deployment/development environment (onprem!) changes to [CHANGELOG.md](https://github.com/reciprocity/zen-ui/blob/main/README.md)

#### Useful links
[Zen UI readme](https://github.com/reciprocity/zen-ui/blob/main/README.md)

[PR review guidelines](https://github.com/reciprocity/zengrc/blob/develop/doc/dev_environment_and_process/pull_request_reviews.md)

[Git commit guidelines](https://github.com/reciprocity/zengrc/blob/develop/doc/dev_environment_and_process/commit_guidelines.md)

### CODE REVIEW

File: src/componets-smart/zen-avatar-group/zen-avatar-group.tsx [hunk 1]
```diff
@@ -41,11 +38,11 @@ export class ZenAvatarGroup {
 
   hiddenUsers(): AvatarData[] {
     const users = this.users.slice();
```
Comment (mplesec) [inline src/componets-smart/zen-avatar-group/zen-avatar-group.tsx:40]: Can this be removed?

File: src/componets-smart/zen-avatar-group/zen-avatar-group.tsx [hunk 2]
```diff
@@ -14,22 +14,19 @@ export class ZenAvatarGroup {
...
-          userName: user.userName,
-          email: user.email,
-          imageUrl: user.imageUrl,
-          color: colors[index].color,
-          background: colors[index].background,
-        };
-      } else {
-        index = 0;
-      }
+      index < colors.length - 1 ? index++ : (index = 0);
```
Comment (mplesec) [inline src/componets-smart/zen-avatar-group/zen-avatar-group.tsx:21]: NIT: This was a confusing line to reason through. Maybe go with `index = index < color.length - 1 ? index + 1 : 0` instead.

Or just simplify it with the modulo operator:
```
let index = 0;
...
index = (index + 1) % colors.length;
```

**Review by mplesec — APPROVED** [review]

File: src/componets-smart/zen-avatar-group/zen-avatar-group.tsx (hunk 1 above)
Comment (mihaelpalfi) [reply src/componets-smart/zen-avatar-group/zen-avatar-group.tsx:40]: I need a new instance of the list. So it dosent change when i am slice ing the list. Dont have lodash clone so this is the old school of doing it right?

Comment (mplesec) [reply src/componets-smart/zen-avatar-group/zen-avatar-group.tsx:40]: Doesn't it already get a new instance with the slice at L:41? Why is is needed twice?

File: src/componets-smart/zen-avatar-group/zen-avatar-group.tsx (hunk 2 above)
Comment (mihaelpalfi) [reply src/componets-smart/zen-avatar-group/zen-avatar-group.tsx]: Nice one. Changed.

File: src/componets-smart/zen-avatar-group/zen-avatar-group.tsx (hunk 1 above)
Comment (mihaelpalfi) [reply src/componets-smart/zen-avatar-group/zen-avatar-group.tsx:40]: Hmm. When i slice it on line 41 directly. Like this.users.splice(this.displayMax, this.users.length). The this.users.length is not correct anymore.

Comment (mplesec) [reply src/componets-smart/zen-avatar-group/zen-avatar-group.tsx:40]: `splice` changes the arr

Comment (mplesec) [reply src/componets-smart/zen-avatar-group/zen-avatar-group.tsx:40]: I'm assuming this is a typo?

Comment (mihaelpalfi) [reply src/componets-smart/zen-avatar-group/zen-avatar-group.tsx]: Aaaaa now i get it.There are slice and splice. Changed!

**Pull request merged**